src/main/java/space/bxteam/ndailyrewards/listeners/PlayerJoinListener.java

```diff
@@ -69,12 +70,17 @@ public void onPlayerJoinEvents(PlayerJoinEvent event) {
         RewardManager rewardManager = NDailyRewards.getInstance().getRewardManager();
         PlayerRewardData playerRewardData = rewardManager.getPlayerRewardData(player.getUniqueId());
         int currentDay = playerRewardData.currentDay() + 1;
+        long DelayTime = NDailyRewards.getInstance().getConfig().getLong("events.auto-claim-delay");
+
 
         if (NDailyRewards.getInstance().getConfig().getBoolean("events.auto-claim-reward") && rewardManager.isRewardAvailable(playerRewardData, currentDay)) {
-            rewardManager.giveReward(player, currentDay);
-            Bukkit.getPluginManager().callEvent(new AutoClaimEvent(player, currentDay));
+            new DelayedTask(() -> {
+                rewardManager.giveReward(player, currentDay);
+                Bukkit.getPluginManager().callEvent(new AutoClaimEvent(player, currentDay));
+            }, DelayTime*20);
```

Comment:
I think it's better to do this, without creating a new class
```suggestion
            new BukkitRunnable() {
                @Override
                public void run() {
                    rewardManager.giveReward(player, currentDay);
                    Bukkit.getPluginManager().callEvent(new AutoClaimEvent(player, currentDay));
                }
            }.runTaskLater(NDailyRewards.getInstance(), delayPeriod * 20L);
```